### ISSUE TITLE
Error creating commit on new git version

### ISSUE THREAD

**Issue by rafelsanso** [stevelacy/gulp-git#180]
I have the last version of Git for Windows **(2.16.2.windows.1)**. With previous versions, I could use the command _git.commit_ without problems. But with this version, I have some issues.
On every gulp task with commit, the process doesn't execute the commit. And this produces errors because I can't push the changes.

I can see a new error on console when try to use commits:
_fatal: empty string is not a valid pathspec. please use . instead if you meant to match all paths_

**Comment by stevelacy** [comment]
Yeah, looks like the 2.16 release changed the warning to an error, ruby-git also hit this: https://github.com/ruby-git/ruby-git/issues/345#issuecomment-369260500

Are you using `git.add` and getting the error, or is it just `git.commit`?

**Comment by rafelsanso** [comment]
This is a simple example of a commit task:

```javascript
// commitFeature
// Save current changes and push to remote
gulp.task('commitFeature', function(){
    var message = argv.message;
    gulp.src('.')
        .pipe(git.add())
        .pipe(git.commit(message), {emitData:true})
        .on('data',function(data) {
            git.push('origin', function (err) {
                if (err) throw err;
            });
        });
});
```

I even use the `git.add` before `git.commit`.

**Comment by lpalli** [comment]
I have the same problem with the last Git version on Windows (2.16.2.windows.1).
Any solution?

**Comment by tgelu** [comment]
+1

**Comment by anoblet** [comment]
+1

**Comment by stevelacy** [comment]
@rafelsanso @anoblet @tgelu @lpalli can you test the latest I pushed to master? Thanks

**Comment by anoblet** [comment]
@stevelacy Sorry I haven't had the time to test, hopefully will be able to soon. Thanks for the work!

**Comment by qhxin** [comment]
+1

**Comment by WilliamStam** [comment]
````
D:\Work\Customer-Care-Admin>npm -g list gulp-git
C:\Users\william.stam\AppData\Roaming\npm
`-- gulp-git@2.6.0
````

still present
````
[12:35:16] Starting 'git-commit'...
[12:35:16]  fatal: empty string is not a valid pathspec. please use . instead if you meant to match all paths
````

````

gulp.task('git-commit', function(done) {
	gitCommitMessage = (typeof gitCommitMessage !== 'undefined' && gitCommitMessage != "") ? gitCommitMessage : "gulp commit";

	var d = new Date();
	var prefix = d.getFullYear() + "-" + ("0" + (d.getMonth() + 1)).slice(-2) + "-" + ("0" + d.getDate()).slice(-2) + " " + ("0" + d.getHours()).slice(-2) + ":" + ("0" + d.getMinutes()).slice(-2) + ":" + ("0" + d.getSeconds()).slice(-2);


	gitCommitMessage = prefix + "\n" + gitCommitMessage;

	git = (typeof git !== 'undefined') ? git : require('gulp-git');

	var timer = duration('git-commit');
	return gulp.src('.')
		.pipe(git.commit(gitCommitMessage))
		.pipe(timer);


});
````

**Comment by stevelacy** [comment]
@WilliamStam can you log out `cmd` on line 89, comment it here and then run that manually?
https://github.com/stevelacy/gulp-git/blob/master/lib/commit.js#L88

**Comment by WilliamStam** [comment]
yikes... sorry! i got busy when i wanted to do it and then sort of forgot about it :( till now
````
[12:45:09] Starting 'git-commit'...
git commit  "" -F - .
[12:45:09]  fatal: empty string is not a valid pathspec. please use . instead if you meant to match all paths
````

**Comment by stevelacy** [comment]
does it work if you move the `.` to before the quotes?

`git commit . "" -F -`
`git commit "" -F . -`

**Comment by WilliamStam** [comment]
the command needs to be `git commit -m "yada" "."` 

the current output is `git commit -m "Latest"   "" .` 

theres somewhere an escape() thats pushing blank "" to the cmd. but i suspect the line 88 should be ` cmd += ' "."';` instead of ` cmd += ' .';` as it is now. just trying to find the "" being added now. but my brain hasnt woken up yet... #mondaymorningblues

**Comment by WilliamStam** [comment]
ok the fix i think...

line 28
````
paths.push(path.relative(opt.cwd, file.path).replace('\\', '/'));
````

becomes 

````
paths.push(path.relative(opt.cwd, file.path).replace('\\', '/')||'.');
````

````
var write = function(file, enc, cb) {
    files.push(file);
    paths.push(path.relative(opt.cwd, file.path).replace('\\', '/')||'.');
    cb();
  };
````

**Comment by stevelacy** [comment]
Want to send a PR with that change?

**Comment by WilliamStam** [comment]
done. hope its ok tho. i haven't tested it if you actually supply a path. but logic says it should work

**Comment by stevelacy** [comment]
@rafelsanso @anoblet @tgelu @lpalli can you try with latest on master?

**Comment by fourthmeal70** [comment]
@stevelacy  I was seeing this same error and Google brought me here. I was able to successfully commit without error using the code on master.  

Please let us know when you release a new version to the NPM registry so I can update my package.json file to point there, instead of the git repo.

**Comment by stevelacy** [comment]
Sounds good.
Published to 2.7.0
